pages/newentry/index.js

```diff
@@ -1,18 +1,15 @@
+import ProtectPage from "@/components/ProtectPages";
 import TransactionEntryForm from "@/components/TransactionEntryForm";
-import { useSession } from "next-auth/react";
 
 function NewEntryPage({ handleAddTransaction }) {
-  const { data: session, status } = useSession();
-
-  if (status !== "authenticated") {
-    return <h2>Access denied!</h2>;
-  }
   return (
     <>
-      <TransactionEntryForm
-        updateTransactions={handleAddTransaction}
-        mode="add"
-      />
+      <ProtectPage>
+        <TransactionEntryForm
+          updateTransactions={handleAddTransaction}
+          mode="add"
+        />
+      </ProtectPage>
```

Comment:
I can't use ProtectPage everywhere and I have to leave the if query just before it on the pages where objects are transferred, otherwise errors arise that I can catch in this way.